subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/generator/TestProjects.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.performance.generator
+
+class TestProjects {
+    static List<String> getProjectMemoryOptions(String testProject) {
+        def daemonMemory = determineDaemonMemory(testProject)
+        return ["-Xms${daemonMemory}", "-Xmx${daemonMemory}"]
+    }
+
+    private static String determineDaemonMemory(String testProject) {
```

Comment:
Why do it like this? Can we embed this information in `JavaTestProject` instead?

Can we create an enum of sorts for these test projects?

Reply:
We can, I opened https://github.com/gradle/gradle/issues/14597 to make this better.